osu.Game/Graphics/UserInterface/ShearedButton.cs

```diff
@@ -216,5 +221,44 @@ private void updateState()
 
             ButtonContent.FadeColour(colourContent, 150, Easing.OutQuint);
         }
+
+        public void AppearFromLeft(double delay)
+        {
+            Content.FinishTransforms();
+            Content.MoveToX(-300f)
+                   .FadeOut()
+                   .Delay(delay)
+                   .MoveToX(0f, 240, Easing.OutCubic)
+                   .FadeIn(240, Easing.OutCubic);
+        }
+
+        public void AppearFromBottom()
+        {
+            Content.FinishTransforms();
+            Content.MoveToY(100f)
+                   .FadeOut()
+                   .MoveToY(0f, 240, Easing.OutCubic)
+                   .FadeIn(240, Easing.OutCubic);
+        }
+
+        public void DisappearToRight(bool expire)
+        {
+            Content.FinishTransforms();
+            Content.FadeOut(240, Easing.InOutCubic)
+                   .MoveToX(300f, 360, Easing.InOutCubic);
+
+            if (expire)
+                this.Delay(Content.LatestTransformEndTime - Time.Current).Expire();
+        }
+
+        public void DisappearToBottom(bool expire)
+        {
+            Content.FinishTransforms();
+            Content.FadeOut(240, Easing.InOutCubic)
+                   .MoveToY(100f, 240, Easing.InOutCubic);
+
+            if (expire)
+                this.Delay(Content.LatestTransformEndTime - Time.Current).Expire();
+        }
```

Comment:
I have so many questions. Like:

- Why is this here?
- Why are transforms being applied to `Content`?
- Why do these have `expire` bool flags?
- Why is this button trying to apply transforms to itself rather than the footer animating it?

I guess `ScreenFooterButton` also does this, which I argue is just as bad? And the fact that this is starting to spread means that it's even worse?

tagging in @frenzibyte since you authored that other stuff

Reply:
I'll answer these set of questions for `ScreenFooterButton`, I don't agree with this existing here.

- In `ScreenFooterButton` context, the button is used exclusively by the footer, so it's either/or whether the animation is defined by the footer or placed in the footer button itself (but it was suggested to put the animation in the button, see https://github.com/ppy/osu/pull/28184#discussion_r1617128386)
- Because the `ScreenFooterButton` may be placed in a fill flow, at which point the animation will be interrupted by the flow layout code. Compare this to the existence of `TopLevelContent` in `ModSelectColumn` to animate it, but I expect you to say it's different because the component is exposing it for the overlay to explicitly apply animation to it, I dunno.
- I suppose you're asking this because I can supposedly do `button.Expire()` from the footer and it'll still expire after the animation is applied, but no, because the animation is applied on `Content` not `this`.
- https://github.com/ppy/osu/pull/28184#discussion_r1617128386

Reply:
Ok well as long as you can delete this stuff from `ShearedButton` I'll look away from `ScreenFooterButton`.